caps/capabilities.go

```diff
@@ -222,3 +236,17 @@ func (e *NotFoundError) Error() string {
 		panic(fmt.Sprintf("unexpected what: %q", e.what))
 	}
 }
+
+// Validate if a capability is correct according to the given type
+func (t Type) Validate(c *Capability) error {
+	if t != c.Type {
+		return fmt.Errorf("capability is not of type %q", t)
+	}
+	// While we don't have any support for type-specific attribute schema,
```

Comment:
We need to look at what the juju team has. They have schema validation already written in Go, and it's indeed based on JSON schema, but it's encoded as yaml, which is likely something we want too. Let's brainstorm on this when we get there.